### PR TITLE
3.x Backport of scope/baggage fix

### LINKED ISSUE #8211: 3.x Backport of #8187 - Telemetry scopes are not closed properly when using baggage
## Environment Details
* Helidon Version: 3.x
* Helidon SE or Helidon MP
* JDK version:
* OS:
* Docker version (if applicable):

----------

## Problem Description
Backport of #8187.

### PR DESCRIPTION
**Pull request by tjquinno**
### Description
Resolves #8211 

Backport of PR #8225 - see the description there for full information.

### Documentation
Bug fix - no doc impact.

### CODE REVIEW

File: tracing/opentelemetry/pom.xml [hunk 1]
```diff
@@ -27,6 +27,10 @@
     <artifactId>helidon-tracing-opentelemetry</artifactId>
     <name>Helidon Tracing Open Telemetry</name>
 
+    <properties>
+        <version.lib.opentelemetry-sdk-extension-autoconfigure>1.22.0-alpha</version.lib.opentelemetry-sdk-extension-autoconfigure>
```
Comment (ljnelson) [inline tracing/opentelemetry/pom.xml:31]: (I assume we're OK with this property at this level, and not in `dependencies/pom.xml` as would normally be the case.)

File: tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    @Override
+    public void forEach(BiConsumer<? super String, ? super BaggageEntry> consumer) {
+        values.forEach(consumer);
+    }
+
+    @Override
+    public Map<String, BaggageEntry> asMap() {
+        return Collections.unmodifiableMap(values);
```
Comment (ljnelson) [inline tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java:52]: Depending on how often this method is called, you may want to store the unmodifiable view as a `final` instance field.

**Review by ljnelson — APPROVED** [review]
Couple of minor comments

File: tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java (hunk 2 above)
Comment (tjquinno) [reply tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java:52]: In the world of the neutral Helidon tracing API, the contents of the baggage data structure can change even once the span is started. So `values` can change at any time.

File: tracing/opentelemetry/pom.xml (hunk 1 above)
Comment (tjquinno) [reply tracing/opentelemetry/pom.xml:31]: My understanding is that for "tangential" dependencies we don't want to load up `dependencies/pom.xml` with them.

Now, maybe OTel is becoming central enough that we do want them there but ideally there should be direct dependencies on OTel only from one (maybe two?) artifacts.

Maybe that's just for integrations and this usage is less tangential than those. 

@tomas-langer @romain-grecourt Thoughts, folks?

File: tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java (hunk 2 above)
Comment (ljnelson) [reply tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java:52]: What I mean is:
```
// in your instance field declarations
private Map<String, BaggageEntry> values; // as before; don't have it in front of me
private Map<String, BaggageEntry> valuesView;

// In your class' constructor:
this.values = new HashMap<>(); // or whatever; all mutation methods target this
this.valuesView = Collections.unmodifiableMap(this.values); // return this from getters
```

Comment (tjquinno) [reply tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java:52]: Ah, yes. Got it. Will change. Was there a reason you didn't advocate for this:

```java
private final Map<String, BaggageEntry> values = new HashMap<>();
private final Map<String, BaggageEntry> valuesView = Collections.unmodifiableMap(values);
```

Comment (ljnelson) [reply tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java:52]: Not particularly; I tend to do all my instance field initializations in the constructor. Style and habit I suppose.

File: tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        return entry != null ? entry.getValue() : null;
+    }
+
+    @Override
+    public BaggageBuilder toBuilder() {
+        return new Builder(values);
+    }
+
+    void baggage(String key, String value) {
+        values.put(key, new HBaggageEntry(value, new HBaggageEntryMetadata("")));
```
Comment (ljnelson) [inline tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java:68]: Not thread safe. OK?

**Review by ljnelson — APPROVED** [review]

File: tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java (hunk 2 above)
Comment (tjquinno) [reply tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/MutableOpenTelemetryBaggage.java]: Changed to save the view.

**Pull request merged**